server.go

```diff
@@ -832,3 +830,33 @@ func (s *Server) syncLocalOrderState() error {
 	// them all in an atomic batch.
 	return s.db.UpdateOrders(noncesToUpdate, ordersToUpdate)
 }
+
+// determineBatchVersion determines the batch version that will be sent to the
+// auctioneer to signal compatibility with the different features the trader
+// client can support.
+func (s *Server) determineBatchVersion() order.BatchVersion {
+	configVersion := s.cfg.DebugConfig.BatchVersion
+
+	// We set the default value of the config flag to -1, so we can
+	// differentiate between no value set and the first version (0).
+	if configVersion >= 0 {
+		return order.BatchVersion(configVersion)
+	}
+
+	isTaprootCompatible := false
+	verErr := lndclient.AssertVersionCompatible(
+		s.lndServices.Version, taprootVersion,
+	)
+	if verErr == nil {
+		isTaprootCompatible = true
+	}
+
+	// We can only auto-upgrade accounts to Taproot if the underlying node
+	// has the required capabilities.
+	if isTaprootCompatible {
+		return order.UpgradeAccountTaprootBatchVersion
+	}
+
+	// Fall back to the previous default version.
+	return order.LatestBatchVersion
```

Comment:
q: can this logic bring us problems in the near future? If we add a new batch version, ex: private channels. We can have a new pool version with private channels but not taproot (using an old lnd version). But we can also have a new pool version with private channels AND taproot :eyes: 

Applies to ` SupportsAccountTaprootUpgrade()` too.